gui/src/2D/controls/inputText.ts

```diff
@@ -923,19 +923,17 @@ export class InputText extends Control implements IFocusableControl {
                 context.fillRect(highlightCursorLeft, this._currentMeasure.top + (this._currentMeasure.height - this._fontOffset.height) / 2, width, this._fontOffset.height);
                 context.globalAlpha = 1.0;
             }
-
-            context.restore();
-
-            // Border
-            if (this._thickness) {
-                if (this.color) {
-                    context.strokeStyle = this.color;
-                }
-                context.lineWidth = this._thickness;
-
-                context.strokeRect(this._currentMeasure.left + this._thickness / 2, this._currentMeasure.top + this._thickness / 2,
-                    this._currentMeasure.width - this._thickness, this._currentMeasure.height - this._thickness);
+        }
+        context.restore();
```

Comment:
This is probably a bug. The restore should be inside the if focused as the associated save is also inside a if focused.

You can test it with the first pg from the initial thread on the forum

Reply:
if I put it inside it creates incomplete border

Reply:
![screenshot 2018-12-07 at 8 10 52 pm](https://user-images.githubusercontent.com/27828189/49653910-52ed0380-fa5c-11e8-97a2-f06defc7aa11.png)
for PG - https://www.babylonjs-playground.com/#YP180Z
@deltakosh - I don't know the cause yet.